Assets/SEE/Layout/NodeLayouts/IncrementalTreeMap/Utils.cs

```diff
@@ -0,0 +1,119 @@
+        }
+
+        /// <summary>
+        /// A list of <paramref name="nodes"/> with rectangles that are laid out
+        /// in a <paramref name="oldRectangle"/> will be transformed (linear), so that they fit
+        /// in <paramref name="newRectangle"/>
+        /// </summary>
+        /// <param name="nodes">nodes with rectangles that should be transformed</param>
+        /// <param name="newRectangle">new parent rectangle</param>
+        /// <param name="oldRectangle">old parent rectangle</param>
+        public static void TransformRectangles(IList<Node> nodes, Rectangle newRectangle, Rectangle oldRectangle)
+        {
+            double scaleX = newRectangle.Width / oldRectangle.Width;
+            double scaleZ = newRectangle.Depth / oldRectangle.Depth;
+
+            foreach (var node in nodes)
+            {
+                node.Rectangle.X = (node.Rectangle.X - oldRectangle.X) * scaleX + newRectangle.X;
+                node.Rectangle.Z = (node.Rectangle.Z - oldRectangle.Z) * scaleZ + newRectangle.Z;
+                node.Rectangle.Width *= scaleX;
+                node.Rectangle.Depth *= scaleZ;
+            }
+        }
+
+        /// <summary>
+        /// Clones a list of nodes and preserves the layout including segments and rectangles.
+        /// Therefor the <see cref="Segment"/>s and <see cref="Rectangle"/>s of the nodes are cloned to.
+        /// </summary>
+        /// <param name="nodes">siblings to be cloned</param>
+        /// <returns>dictionary that maps ID to new clone node</returns>
+        public static IDictionary<string, Node> CloneGraph(IList<Node> nodes)
+        {
+            // clones the nodes only partially: does NOT clone the segment
+            IDictionary<string, Node> clonesMap = nodes.ToDictionary(node => node.ID,
+                node => new Node(node.ID)
+                {
+                    Rectangle = node.Rectangle.Clone(), Size = node.Size
+                }
+            );
+            CloneSegments(nodes, clonesMap);
+            return clonesMap;
+        }
+
+        /// <summary>
+        /// Clones all <see cref="Segment"/>s of <paramref name="from"/>
+        /// and register the segment clones to the nodes of <paramref name="to"/>.
+        /// The nodes of <paramref name="from"/> and <paramref name="to"/> needs to have the same IDs.
+        /// </summary>
+        /// <param name="from">nodes with segment structure, that should be copied</param>
+        /// <param name="to">dictionary that maps ID to nodes, that should get the segment structure </param>
+        public static void CloneSegments(IEnumerable<Node> from, IDictionary<string, Node> to)
+        {
+            var segments = from.SelectMany(n => n.SegmentsDictionary().Values).ToHashSet();
+            foreach (var segment in segments)
+            {
+                var segmentClone = new Segment(segment.IsConst, segment.IsVertical);
+                foreach (var node in segment.Side1Nodes.ToList())
```

Comment:
It's not called to convert it to list, but to create a copy.
This is necessary because the actual list could change while iterating over it, if the arguments ```form``` and ```to``` contain identical nodes.
The case would be like 'clone the segments from x to x'. We can of cause also avoid to call this function for this case, would this be a better practice?

Reply:
Ah, I see, thanks for the explanation. The existing code is fine, but it may be a good idea to add a short comment above the line so it's clear why the `ToList` happens.

Reply:
Commented in 1c05bafa2d15e2ec4c76f0cf39925169d7a12a8a.